compiler/src/iree/compiler/Dialect/HAL/IR/test/tensor_folding.mlir

```diff
@@ -40,6 +40,45 @@ util.func public @foldTensorExportImport(%arg0: tensor<5xi32>) -> tensor<5xi32>
 
 // -----
 
+// CHECK-LABEL: @FoldConsecutiveTransientsSameStorage
+// CHECK-SAME: (%[[ARG0:.+]]: tensor<4xf32>, %[[STORAGE:.+]]: !hal.buffer)
+util.func public @FoldConsecutiveTransientsSameStorage(%arg0: tensor<4xf32>, %storage: !hal.buffer) -> tensor<4xf32> {
+  // CHECK-NOT: hal.tensor.transients
+  %0 = hal.tensor.transients %arg0 : tensor<4xf32> from %storage : !hal.buffer
```

Comment:
Im curious why you think this will ever happen. I expect this to be generated by front end and I am not really sure I even understand what it means for a `hal.tensor.transient` to have it source be another `hal.tensor.transient`. Shouldnt that just be an error?